src/main.ts

```diff
@@ -194,6 +202,19 @@ export default class ObsidianDiscordRPC extends Plugin {
           largeImageKey: this.settings.themeStyle,
           largeImageText: "no info just privacy mode",
         });
+      } else if (
+        this.settings.showVaultName &&
+        this.settings.showCurrentFileName &&
+        this.settings.showFolderName &&
+        folderPath
+      ) {
+        await this.rpc.setActivity({
+          details: `Editing ${file}`,
+          state: `Vault: ${vault}  ▸ ${folderPath}`,
+          startTimestamp: date,
+          largeImageKey: this.settings.themeStyle,
+          largeImageText: "I'm thinking!",
+        });
```

Comment:
```suggestion
      } else if (
        this.settings.showVaultName &&
        this.settings.showCurrentFileName &&
        this.settings.showFolderName &&
        folderPath
      ) {
        await this.rpc.setActivity({
          details: `Editing: ${folderPath} ▸ ${file}`,
          state: `Vault: ${vault}`,
          startTimestamp: date,
          largeImageKey: this.settings.themeStyle,
          largeImageText: "I'm thinking!",
        });
```
here is my suggestion

but maybe the folder will have a long name and the note title will not show